kv/kvclient/kvcoord: TestWaiterOnRejectedCommit failed

kv/kvclient/kvcoord.TestWaiterOnRejectedCommit [failed](https://tanzanite.cluster.engflow.com/invocations/default/0ff72fd1-1559-431e-b0f1-b281ce3c0765?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [507c68f8abb0c9a3f527c66cb42129c2cb096e7d](https://github.com/cockroachdb/cockroach/commits/507c68f8abb0c9a3f527c66cb42129c2cb096e7d):

Fatal error:

```
panic: test timed out after 53m20s
running tests:
	TestWaiterOnRejectedCommit (53m12s)

```
Stack: 

```
goroutine 6132 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* 
* goroutine 4108 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	pkg/util/admission/work_queue.go:501 +0x1cd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 4062
* 	pkg/util/admission/work_queue.go:472 +0x8e
* 
* goroutine 4109 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	pkg/util/admission/work_queue.go:424 +0xbd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 4062
* 	pkg/util/admission/work_queue.go:421 +0x676
* 
* goroutine 4110 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	pkg/util/admission/work_queue.go:501 +0x1cd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 4062
* 	pkg/util/admission/work_queue.go:472 +0x8e
* 
* goroutine 4086 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*cleanupManager).mainLoop(0xc005180640)
* 	external/com_github_cockroachdb_pebble/obsolete_files.go:164 +0x1e7
* github.com/cockroachdb/pebble.openCleanupManager.func1.1({0xbb24438?, 0xc00167ed80?})
* 	external/com_github_cockroachdb_pebble/obsolete_files.go:99 +0x2f
* runtime/pprof.Do({0xbb24108, 0x10a4e980}, {{0xc000e5d500?, 0xc0000bbfd0?, 0x78f8e5?}}, 0xc0000bbfc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble.openCleanupManager.func1()
* 	external/com_github_cockroachdb_pebble/obsolete_files.go:98 +0x92
* created by github.com/cockroachdb/pebble.openCleanupManager in goroutine 4062
* 	external/com_github_cockroachdb_pebble/obsolete_files.go:97 +0x545
* 
* goroutine 4111 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	pkg/util/admission/work_queue.go:424 +0xbd
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 4062
* 	pkg/util/admission/work_queue.go:421 +0x676
* 
* goroutine 6115 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.watchForNotifications({0xbb24400, 0xc00576eb90}, 0xc00219b150, 0xc005a1ce40)
* 	pkg/kv/kvserver/concurrency/lock_table_waiter.go:829 +0xaf
* created by github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.(*lockTableWaiterImpl).WaitOn.func3 in goroutine 6083
* 	pkg/kv/kvserver/concurrency/lock_table_waiter.go:322 +0x1e5
* 
* goroutine 6114 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.watchForNotifications({0xbb24400, 0xc00576e910}, 0xc00219aac0, 0xc0021a0cc0)
* 	pkg/kv/kvserver/concurrency/lock_table_waiter.go:829 +0xaf
* created by github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency.(*lockTableWaiterImpl).WaitOn.func3 in goroutine 6084
* 	pkg/kv/kvserver/concurrency/lock_table_waiter.go:322 +0x1e5
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWaiterOnRejectedCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47116

--- a/pkg/kv/kvclient/kvcoord/integration_test.go
+++ b/pkg/kv/kvclient/kvcoord/integration_test.go
@@ -62,7 +62,7 @@ func TestWaiterOnRejectedCommit(t *testing.T) {
 	txnUpdate := make(chan roachpb.TransactionStatus, 20)
 
 	illegalLeaseIndex := true
-	s, _, db := serverutils.StartServer(t, base.TestServerArgs{
+	s, sqlDB, db := serverutils.StartServer(t, base.TestServerArgs{
 		Knobs: base.TestingKnobs{
 			Store: &kvserver.StoreTestingKnobs{
 				DisableMergeQueue: true,
@@ -133,6 +133,16 @@ func TestWaiterOnRejectedCommit(t *testing.T) {
 		},
 	})
 	defer s.Stopper().Stop(ctx)
+
+	// Disable parallel commits protocol. In some rare cases, we see two different
+	// command ids for the committing this transaction. The first one could be the
+	// one through parallel commits protocol and the second is using the normal
+	// commit protocol. There is a test race where TestingProposalFilter changes
+	// the commit command id from the first one (parallel commit) to the second
+	// one (normal commit). This causes the test to inject the error on the wrong
+	// command id. Disabling parallel commits avoids this test race.
+	_, err := sqlDB.Exec("SET CLUSTER SETTING kv.transaction.parallel_commits_enabled = 'false'")
+	require.NoError(t, err)
 
 	if _, _, err := s.SplitRange(roachpb.Key("b")); err != nil {
 		t.Fatal(err)

